src/simulation/dynamics/constraintEffector/constraintDynamicEffector.h

```diff
@@ -0,0 +1,121 @@
+/*
+ ISC License
+
+ Copyright (c) 2024, Autonomous Vehicle Systems Lab, University of Colorado at Boulder
+
+ Permission to use, copy, modify, and/or distribute this software for any
+ purpose with or without fee is hereby granted, provided that the above
+ copyright notice and this permission notice appear in all copies.
+
+ THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+ WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+ ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+ OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+
+ */
+
+
+#ifndef CONSTRAINT_DYNAMIC_EFFECTOR_H
+#define CONSTRAINT_DYNAMIC_EFFECTOR_H
+
+#include "simulation/dynamics/_GeneralModuleFiles/dynamicEffector.h"
+#include "simulation/dynamics/_GeneralModuleFiles/stateData.h"
+#include "architecture/_GeneralModuleFiles/sys_model.h"
+
+#include "architecture/messaging/messaging.h"
+
+#include "architecture/utilities/bskLogging.h"
+#include "architecture/utilities/avsEigenMRP.h"
+#include <Eigen/Dense>
+#include <vector>
+// #include <iostream>
+// #include <cstring>
+
+/*! @brief constraint dynamic effector class */
+class ConstraintDynamicEffector: public SysModel, public DynamicEffector {
+public:
+    ConstraintDynamicEffector();
+    ~ConstraintDynamicEffector();
+    void Reset(uint64_t CurrentSimNanos);
+    void linkInStates(DynParamManager& states);
+    void computeForceTorque(double integTime, double timeStep);
+    void UpdateState(uint64_t CurrentSimNanos);
+
+    /** setter for `r_P2P1_B1Init` initial spacecraft separation */
+    void setR_P2P1_B1Init(Eigen::Vector3d r_P2P1_B1Init);
+    /** setter for `r_P1B1_B1` connection point position on spacecraft 1 */
+    void setR_P1B1_B1(Eigen::Vector3d r_P1B1_B1);
+    /** setter for `r_P2B2_B2` connection point position on spacecraft 2 */
+    void setR_P2B2_B2(Eigen::Vector3d r_P2B2_B2);
+    /** setter for `alpha` gain tuning parameter */
+    void setAlpha(double alpha);
+    /** setter for `beta` gain tuning parameter */
+    void setBeta(double beta);
+    /** setter for `k_d` gain */
+    void setKd(double k_d);
```

Comment:
The module variable logging only works if the getters use the exact same name as the variable.  For example, if the variable is `k_d`, the the logging code expects `getK_d()` to return the `k_d` variable.  Right now you are using slightly different names for the setters and getters than the variable names.  Can you please make this consistent?  Very quick fix.